[BUG] npm run check fails on main — auth.ts references undefined hashPassword/verifyPassword functions

**Describe the bug**
Running `npm run check` (tsc) on `main` fails with 4 TypeScript errors in `server/auth.ts`. Commit `3e0a873` (fix: hash registered passwords #289) added calls to `hashPassword()` and `verifyPassword()` and a `passwordHash` property on the `RegisteredUser` interface, but these were never defined or imported. The type check on `main` is completely broken.

**To Reproduce**
Steps to reproduce the behavior:
1. `git checkout main && git pull`
2. `npm install`
3. `npm run check`
4. See errors:
```
server/auth.ts(87,7): error TS2561: Object literal may only specify known properties, but 'passwordHash' does not exist in type 'RegisteredUser'. Did you mean to write 'password'? server/auth.ts(87,21): error TS2552: Cannot find name 'hashPassword'. Did you mean 'password'? server/auth.ts(120,29): error TS2304: Cannot find name 'verifyPassword'. server/auth.ts(120,69): error TS2551: Property 'passwordHash' does not exist on type 'RegisteredUser'. Did you mean 'password'?

```

**Expected behavior**
`npm run check` should pass with zero errors on `main`.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: All (CI / local)

**Additional context**
Introduced by commit `3e0a873` (#289) which attempted to fix issue #288 (plaintext passwords). The fix added `hashPassword()` and `verifyPassword()` calls in the registration and login handlers, and changed the `RegisteredUser` interface usage to `passwordHash` instead of `password`, but the utility functions and interface update were never completed. The `RegisteredUser` interface at line 19 still uses `password: string` instead of `passwordHash: string`.

--- a/server/auth.ts
+++ b/server/auth.ts
@@ -109,11 +109,6 @@ export function createAuthRouter(): Router {
       return res.status(400).json({ message: "Password must be at least 8 characters." });
     }
 
-    const existing = await storage.getUserByEmail(email);
-    if (existing) {
-      return res.status(409).json({ message: "An account with this email already exists." });
-    }
-
     // Check DB for existing user
     try {
       const db = getDb();
@@ -126,12 +121,6 @@ export function createAuthRouter(): Router {
       if (existingDbUser) {
         return res.status(409).json({ message: "An account with this email already exists." });
       }
-    registeredUsers.set(email, {
-      fullName,
-      email,
-      passwordHash,
-      medicalLicenseNumber: licenseNumber,
-    });
 
       // Create DB user
       const passwordHash = hashPassword(password);
@@ -193,29 +182,19 @@ export function createAuthRouter(): Router {
     if (email === devEmail && password === devPassword) {
       userFullName = "Dr. Smith";
     } else {
-      // Check in-memory store (legacy)
-      const registeredUser = registeredUsers.get(email);
-      if (registeredUser && bcrypt.compareSync(password, registeredUser.password)) {
-        userName = registeredUser.fullName;
-      }
+      try {
+        const db = getDb();
+        const [dbUser] = await db
+          .select()
+          .from(users)
+          .where(eq(users.email, email))
+          .limit(1);
 
-      // Also check DB
-      if (!userName) {
-        try {
-          const db = getDb();
-          const [dbUser] = await db
-            .select()
-            .from(users)
-            .where(eq(users.email, email))
-            .limit(1);
-
-          if (dbUser && verifyPassword(password, dbUser.passwordHash)) {
-            userName = dbUser.fullName;
-          }
-        } catch (err) {
-          // DB not available — fall back to in-memory only
-          console.warn("DB unavailable for login, using in-memory only.");
+        if (dbUser && verifyPassword(password, dbUser.passwordHash)) {
+          userFullName = dbUser.fullName;
         }
+      } catch (err) {
+        console.warn("DB unavailable for login.");
       }
     }
 
@@ -274,7 +253,12 @@ export function createAuthRouter(): Router {
       name = "Dr. Smith";
       id = "dev";
     } else {
-      const user = await storage.getUserByEmail(email);
+      const db = getDb();
+      const [user] = await db
+        .select()
+        .from(users)
+        .where(eq(users.email, email))
+        .limit(1);
       if (!user) {
         return res.status(404).json({ message: "User not found." });
       }
@@ -327,7 +311,7 @@ export function createAuthRouter(): Router {
 
       // If already verified, return success
       if (user.emailVerified) {
-        req.session.user = { email: user.email, name: user.fullName };
+        req.session.user = { id: user.id, email: user.email, name: user.fullName };
         return res.json({ success: true, message: "Email already verified." });
       }
 
@@ -391,7 +375,7 @@ export function createAuthRouter(): Router {
         .where(eq(users.id, user.id));
 
       // Create session
-      req.session.user = { email: user.email, name: user.fullName };
+      req.session.user = { id: user.id, email: user.email, name: user.fullName };
 
       return res.json({ success: true, message: "Email verified successfully." });
     } catch (err) {

--- a/server/middleware/rateLimiter.ts
+++ b/server/middleware/rateLimiter.ts
@@ -6,11 +6,11 @@ const MAX_REQUESTS = 60;
 const MAX_TRACKED_CLIENTS = 1000;
 
 function pruneExpiredRecords(now: number) {
-  for (const [clientId, record] of requestStore.entries()) {
+  requestStore.forEach((record, clientId) => {
     if (now > record.resetTime) {
       requestStore.delete(clientId);
     }
-  }
+  });
 }
 
 function trimRequestStore(now: number) {

--- a/server/storage.ts
+++ b/server/storage.ts
@@ -8,7 +8,7 @@ import {
   type User,
   type InsertUser
 } from "@shared/schema";
-import { desc, eq } from "drizzle-orm";
+import { and, desc, eq } from "drizzle-orm";
 
 export interface IStorage {
   getAssessments(limit?: number, offset?: number, createdBy?: string): Promise<Assessment[]>;
@@ -35,13 +35,18 @@ export class DatabaseStorage implements IStorage {
   ): Promise<Assessment[]> {
     const db = getDb();
 
-    let query = db
+    const filters: any[] = [];
+    if (createdBy) {
+      filters.push(eq(assessments.createdBy, createdBy));
+    }
+
+    const query = db
       .select()
       .from(assessments)
       .orderBy(desc(assessments.createdAt));
 
-    if (createdBy) {
-      query = query.where(eq(assessments.createdBy, createdBy));
+    if (filters.length > 0) {
+      return await query.where(and(...filters)).limit(limit).offset(offset);
     }
 
     return await query.limit(limit).offset(offset);

